python/vyos/vpp/utils.py

```diff
@@ -77,6 +80,21 @@ def cli_ifaces_list(config_instance, mode: str = 'candidate') -> list[str]:
     for iface in config.get('settings', {}).get('interface', {}).keys():
         vpp_ifaces.append(iface)
 
+    # Add Ethernet interfaces with vifs
+    if with_vifs:
+        from vyos.configdict import get_interface_dict
+
+        for iface in vpp_ifaces:
+            _, iface_config = get_interface_dict(
+                config_instance, ['interfaces', 'ethernet'], ifname=iface
+            )
+            vpp_ifaces.extend([f'{iface}.{vif}' for vif in iface_config.get('vif', {})])
+            vpp_ifaces.extend(
+                [f'{iface}.{vif_s}' for vif_s in iface_config.get('vif_s', {})]
+            )
+
+        return vpp_ifaces
+
```

Comment:
I would prefer to make it a dedicated function. If combined it is confusing, because a list `with_vifs=True` excludes other VPP interfaces, and turns the whole function to some `ethernet_with_vifs`, which is not obvious.